### PR TITLE
feat(icon): add label property for a11y

### PR DESCRIPTION
**Pull request by jcfranco**
This adds `textLabel` (convenience property) to `calcite-icon` for accessibility.

Note: this also updates the `commonTests` util (same as `calcite-app-components` without the dependency-loading options).

#286

### CODE REVIEW

**Comment by macandcheese** [comment]
Would it be easier to just spread any attributes provided to the Host element, so implementers can use `aria-label` directly like `<calcite-icon icon="chevron" scale="s" aria-label="my label"></calcite-icon>`?

**Comment by jcfranco** [comment]
@macandcheese I don't think spreading the attributes is necessary. Although, I'm reconsidering this property since it's now just an alias for `aria-label`.

After a few iterations, I'd like to ask @Esri/calcite-components whether we want `calcite-icon` to:

1. set `role=img` and expect user to set `aria-label`, `aria-labelledby` or `title
  
    This approach forces users  to explicitly label the element or set `role=presentation` if the icon is labeled elsewhere.

2. don't set a role and let users choose their own adventure

    This approach won't nag users, but at the same time will not raise flags when the icon is meant to be meaningful.

**Note**: ☝️ assumes users are doing some sort of a11y check

**Comment by julio8a** [comment]
https://fontawesome.com/how-to-use/on-the-web/other-topics/accessibility

**Comment by jcfranco** [comment]
I updated the component to add semantic-icon attributes when `textLabel` is provided. Otherwise, the icon will be treated as decorative. I had to set  props on host when it's semantic due to the shadow DOM (the accessibility tree didn't seem to pick up any info from the shadow DOM, at least from axe + VoiceOver testing).

File: src/tests/commonTests.ts [hunk 1]
```diff
@@ -1,46 +1,70 @@
...
+}
+
+export async function accessible(componentTagOrHTML: TagOrHTML): Promise<void> {
+  const page = await simplePageSetup(componentTagOrHTML);
+  await page.addScriptTag({ path: require.resolve("axe-core") });
+
+  expect(
+    await page.evaluate(
+      async (componentTag: CalciteComponentTag) =>
+        (window as AxeOwningWindow & typeof globalThis).axe.run(componentTag),
```
Comment (paulcpederson) [inline src/tests/commonTests.ts]: this is very nifty

**Comment by paulcpederson** [comment]
:100:

**Pull request merged**